swap/src/protocol/alice/event_loop.rs

```diff
@@ -123,13 +127,27 @@ impl EventLoopHandle {
         Ok(())
     }
 
-    pub async fn send_transfer_proof(&mut self, bob: PeerId, msg: TransferProof) -> Result<()> {
+    pub async fn send_transfer_proof(
+        &mut self,
+        bob: PeerId,
+        msg: TransferProof,
+        execution_params: ExecutionParams,
+    ) -> Result<()> {
         let _ = self.send_transfer_proof.send((bob, msg)).await?;
 
-        self.recv_transfer_proof_ack
-            .recv()
-            .await
-            .ok_or_else(|| anyhow!("Failed to receive transfer proof ack from Bob"))?;
+        // TODO: Re-evaluate if these acknowledges are necessary at all.
```

Comment:
I commented on the review re-ack. I think they are necessary to avoid scenarios where one party restart and wrongly assumes that the previous message they sent did send.

Are you saying that if Alice did not dial Bob then the `send_transfer_proof.send` does **not** fail?

Reply:
That is exactly what I am saying. Without the timeout the program hangs forever on awaiting the ack.